src/index/CompressedRelationPermutationWriterImpl.h

```diff
@@ -155,8 +153,24 @@ struct CompressedRelationWriter::PermutationWriter {
     AD_CORRECTNESS_CHECK(numColumns_ == writer2_.numColumns());
 
     writer1_.smallBlocksCallback_ =
-        AddBlockOfSmallRelationsToSwitched<Writer2CompressAndWriteBlock>{
-            Writer2CompressAndWriteBlock{writer2_}};
+        AddBlockOfSmallRelationsToSwitched{writer2_};
+  };
+
+  // Constructor for a `PermutationWriter` which writes a single permutation.
+  PermutationWriter(WriterAndCallback writerAndCallback1,
+                    qlever::KeyOrder permutation,
+                    PerBlockCallbacks perBlockCallbacks) requires(!WritePair)
+      : permutation_{std::move(permutation)},
+        writer1_{writerAndCallback1.writer_},
+        writeMetadata_{std::move(writerAndCallback1.callback_),
+                       writerAndCallback1.writer_.blocksize()},
+        blocksize_{writerAndCallback1.writer_.blocksize()},
+        numColumns_{writerAndCallback1.writer_.numColumns()},
+        relation_{numColumns_, alloc_},
```

Comment:
I think that those three can be directly initialized where the members are defined (this works if they only depend on members that are declared before them etc., 
then you can get rid of some duplication between the two constructors.